### ISSUE TITLE
Update watermark with commands relevant in context

### ISSUE THREAD

**Issue by chrmarti** [Microsoft/vscode#15547]
We should remove the editor commands from the watermark because these cannot be acted on at the time the watermark is shown. We also need to distinguish between the workbench being open on a folder or not, some commands make more sense in one case than the other. The following is a proposal for the (currently 5) commands in each case:

Commands in an empty workbench:
- Show All Commands
- Open File
- Open Folder (not on Mac where it is included in Open File)
- Open Recent
- New File (only on Mac)
- Toggle Terminal

Commands in a workbench opened on a folder:
- Show All Commands
- Go to File (QuickOpen)
- Search in Files
- Start Debugging
- Toggle Terminal

Another distinct context is that of the first session after installation where the user is looking for ways to explore and become comfortable with the workbench.

Commands in first session after install:
- Show All Commands
- Choose Theme
- Choose Keymap
- Open Folder
- Keyboard Cheatsheet

/cc @bpasero @joaomoreno @sandy081 @chrisdias @seanmcbreen @kieferrm @bgashler1 for feedback

Plan item: #15383

**Comment by bpasero** [comment]
@chrmarti I also suggest to disable the watermark experiment and show the watermark for all users.

**Comment by chrmarti** [comment]
Any objection towards not opening an untitled file in the empty workbench? The corresponding experiment doesn't show negative impact of not showing it, but not showing it increases the usage of commands in the watermark.

**Comment by bpasero** [comment]
@chrmarti I was only referring to having the watermark always visible when the last editor is closed. I would not change the default of opening an untitled file or not until we found out that this changes something for our first time users. From what I heard so far, our experiments did not improve discoverability yet.

**Comment by chrmarti** [comment]
@bpasero This is an update on that: The experiment shows that usage of the commands in the watermark increases and not showing the untitled file increases usage further. What we don't see yet is this translating to a higher retention rate, but that might be a matter of picking the right commands and combining this measure with others to improve discoverability and usability overall.

**Comment by bpasero** [comment]
@chrmarti on that topic I would prefer to explore the possibility of opening an untitled file and a welcome editor in a second group to make first time users discover our commands.

Are you suggesting to not open an untitled file only for first time users (once) or in general?

**Comment by chrmarti** [comment]
@bpasero I'm suggesting to not open an untitled file for both first time and returning users. Note that this only affects the empty workbench case. I don't see an effect of the untitled file experiment on the number of times an untitled file is used. It appears that creating an untitled file is intuitively understood by new users. Also users don't create new files that frequently, they read/work on existing files much more often it seems, that reduces the value of the automatically opened new untitled file further.

**Comment by bpasero** [comment]
@chrmarti got it. my personal opinion is that we should not start an editor instance without showing an editor. in the end that is what we are (an editor) and many people might expect that they can just start typing.

This becomes especially more important once hot exit is enabled by default because then you can use Code as notebook by quickly opening it, typing something into the untitled buffer and closing it without saving.

**Comment by chrmarti** [comment]
@bpasero I think we have a gap with regard to the no-folder-opened case in that we don't restore previously opened files (#207) and implementing that would significantly reduce the number of cases where the workbench opens empty (or with the new untitled file).

I can see Code being used as a notebook, but I guess most users will feel more comfortable with a file that has a name and is stored in a proper folder than a dirty untitled file using hot-exit. I'm using Code as a notebook myself by opening it on a non-project folder to get the files reopened after restart and have auto-save enabled, so I don't need to worry about that.

**Comment by chrmarti** [comment]
Keeping the new untitled file experiment in for new users (extended to the first day after installation like the watermark for new users). Will open a new issue to follow up on that.
